src/sources/mongodb_metrics/mod.rs

```diff
@@ -0,0 +1,1065 @@
+    #[snafu(display("failed to parse `buildInfo` response: {:?}", source))]
+    CommandBuildInfoParseError { source: bson::de::Error },
+    #[snafu(display("only `Mongod` supported right now, current: `{:?}`", node_type))]
+    UnsupportedNodeType { node_type: NodeType },
+}
+
+#[derive(Debug)]
+enum CollectError {
+    Mongo(MongoError),
+    Bson(bson::de::Error),
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, Default)]
+#[serde(deny_unknown_fields)]
+struct MongoDBMetricsConfig {
+    endpoint: String,
+    #[serde(default = "default_scrape_interval_secs")]
+    scrape_interval_secs: u64,
+    #[serde(default = "default_namespace")]
+    namespace: String,
+}
+
+#[derive(Debug)]
+struct MongoDBMetrics {
+    client: Client,
+    endpoint: String,
+    namespace: String,
+    tags: BTreeMap<String, String>,
+}
+
+pub fn default_scrape_interval_secs() -> u64 {
+    15
+}
+
+pub fn default_namespace() -> String {
+    "mongodb".to_string()
+}
+
+inventory::submit! {
+    SourceDescription::new::<MongoDBMetricsConfig>("mongodb_metrics")
+}
+
+impl_generate_config_from_default!(MongoDBMetricsConfig);
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "mongodb_metrics")]
+impl SourceConfig for MongoDBMetricsConfig {
+    async fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        shutdown: ShutdownSignal,
+        out: Pipeline,
+    ) -> crate::Result<super::Source> {
+        let mongodb = MongoDBMetrics::new(&self.endpoint, &self.namespace).await?;
+
+        let out = out
+            .sink_map_err(|e| error!("error sending metric: {:?}", e))
+            .sink_compat();
+
+        let task = interval(Duration::from_secs(self.scrape_interval_secs))
+            .take_until(shutdown.compat())
+            .filter_map(move |_| {
+                let mongodb = Arc::clone(&mongodb);
+                async { Some(mongodb.collect().await) }
+            })
+            .flatten()
+            .map(Event::Metric)
+            .map(Ok)
+            .forward(out)
+            .inspect(|_| info!("finished sending"));
+
+        Ok(Box::new(task.boxed().compat()))
+    }
+
+    fn output_type(&self) -> crate::config::DataType {
+        config::DataType::Metric
+    }
+
+    fn source_type(&self) -> &'static str {
+        "mongodb_metrics"
+    }
+}
+
+impl MongoDBMetrics {
+    /// Works only with Standalone connection-string. Collect metrics only from specified instance.
+    /// https://docs.mongodb.com/manual/reference/connection-string/#standard-connection-string-format
+    async fn new(endpoint: &str, namespace: &str) -> Result<Arc<MongoDBMetrics>, BuildError> {
+        let mut tags: BTreeMap<String, String> = BTreeMap::new();
+
+        let mut client_options = ClientOptions::parse(endpoint)
+            .await
+            .context(InvalidEndpoint)?;
+        client_options.direct_connection = Some(true);
+        tags.insert("host".into(), client_options.hosts[0].hostname.clone());
```

Comment:
I think we want to include the port here too. We did that for `apache_metrics`: https://github.com/timberio/vector/blob/master/rfcs/2020-08-21-3092-apache-metrics-source.md